test/test_linalg.py

```diff
@@ -7522,15 +7532,14 @@ def run_test(A_dims, b_dims):
     @skipCPUIfNoLapack
     @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
     def test_lu_solve_batched_broadcasting(self, device, dtype):
-        from numpy.linalg import solve
         from torch.testing._internal.common_utils import random_fullrank_matrix_distinct_singular_value
 
         def run_test(A_dims, b_dims, pivot=True):
             A_matrix_size = A_dims[-1]
             A_batch_dims = A_dims[:-2]
-            A = random_fullrank_matrix_distinct_singular_value(A_matrix_size, *A_batch_dims, dtype=dtype)
+            A = random_fullrank_matrix_distinct_singular_value(A_matrix_size, *A_batch_dims, dtype=dtype, device=device)
             b = torch.randn(*b_dims, dtype=dtype)
-            x_exp = torch.as_tensor(solve(A.numpy(), b.numpy())).to(dtype=dtype, device=device)
+            x_exp = np.linalg.solve(A.numpy(), b.numpy())
```

Comment:
I think this is causing errors on ROCm

```
16:08:34 ======================================================================
16:08:34 ERROR [0.071s]: test_lu_solve_batched_broadcasting_cuda_float32 (__main__.TestLinalgCUDA)
16:08:34 ----------------------------------------------------------------------
16:08:34 Traceback (most recent call last):
16:08:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1117, in wrapper
16:08:34     method(*args, **kwargs)
16:08:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 373, in instantiated_test
16:08:34     result = test(self, **param_kwargs)
16:08:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 801, in dep_fn
16:08:34     return fn(slf, *args, **kwargs)
16:08:34   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 801, in dep_fn
16:08:34     return fn(slf, *args, **kwargs)
16:08:34   File "test_linalg.py", line 7549, in test_lu_solve_batched_broadcasting
16:08:34     run_test((2, 1, 3, 4, 4), (2, 1, 3, 4, 6))  # no broadcasting
16:08:34   File "test_linalg.py", line 7542, in run_test
16:08:34     x_exp = np.linalg.solve(A.numpy(), b.numpy())
16:08:34 TypeError: can't convert cuda:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.
```

I'm not sure why it's not causing an error on other jobs? Shouldn't this be failing on every CUDA job?

Reply:
I check the list of CI jobs and it seems like a bunch of CUDA jobs were skipped, and the ones that were running didn't run the tests from `test_linalg.py` at all. Leaving only the ROCm to be the only "CUDA" CI job. Very interesting situation! This test should and would have failed in CUDA jobs.
Maybe the reason I use the new CI Flow the default set of PRs is now different for me and I need to explicitly ask for CUDA builds.

Reply:
hm... that's worrying. cc @zhouzhuojie to take a look at this

Reply:
> Many CUDA builds were skipped, I added the `ciflow/cuda` label so that they are run now.

I would think CUDA and test_linalg.py would be run by default, though. cc @zhouzhuojie